File: imagemagick/output.go
package imagemagick

import "fmt"

type OutputNode struct {
	filename string
}

func (node *OutputNode) Typeid() uint {
	return NODE_TYPE_OUTPUT
}

func (node *OutputNode) GetArgs() string {
	args := fmt.Sprintf("%s", node.filename)

	return args
}

File: src/imagemagick/output.rs
use super::global::NODE_TYPE_OUTPUT;
use super::node::Node;
use derives::node_manager;

#[node_manager]
pub struct OutputNode {
    filename : String
}

impl Node for OutputNode {
    fn type_id(& self) -> u32 {
        return NODE_TYPE_OUTPUT;
    }
    fn get_args(self: & OutputNode) -> Option<String> {
        return match self.filename.len() {
            0 => None,
            _ => Some(self.filename.clone())
        };
    }
}